### PR TITLE
Add file Lesson_07/homework_07.py

### PR DESCRIPTION
**Pull request by barchiso**
Add homework_07

Tasks #7 -10
Use tasks from previous homework or bonus tasks.

### CODE REVIEW

File: Lesson_07/homework_07.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    Args:
+        str1 (string): First string
+        str2 (string): Second string
+
+    Returns:
+        int: Index of the first entry
+    """
+    if str2 in str1:
+        return str1.find(str2[0])
+    return -1
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]: Incorrect cause here you just check first letter + could be optimized in one line. 
Try do not invent cycle  - just read str.find  documentation

File: Lesson_07/homework_07.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        numbers_list (list): List of numbers.
+
+    Returns:
+        int: Sum of even numbers.
+    """
+    even_numbers_sum = 0
+    for number in numbers_list:
+        if number % 2 == 0:
+            even_numbers_sum += number
+    return even_numbers_sum
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]:
```suggestion
    even_numbers_sum = 0
    for number in numbers_list:
        if number % 2 == 0:
            even_numbers_sum += number
    return even_numbers_sum
```
Please try to use list comprehension here + sum()

File: Lesson_07/homework_07.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    """Date formatted string.
+
+    Args:
+        args (tuple): Tuple with three inputs: Day, Month, Year
+        sep (str, optional): _description_. Defaults to ".".
+
+    Returns:
+        string: date in formatted string
+    """
+    date_string = sep.join(map(str, args))
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]: Please try to re-write the same with list comprehension and without map

File: Lesson_07/homework_07.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    Args:
+        number (integer): any number
+
+    Returns:
+        boolean: True or False
+    """
+    for i in range(2, int(number ** 0.5) + 1):
+        if number % i == 0:
+            return False
+        return True
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]:
```suggestion
    for i in range(2, int(number ** 0.5) + 1):
        if number % i == 0:
            return False
        return True
```
print(is_prime_result) -> True ( but 9 is not prime number) - please recheck your algorithm 
Pay attention also to code when return boolean values - usually these conditions could be optimized

**Review by antohaUa — APPROVED** [review]

File: Lesson_07/homework_07.py (hunk 4 above)
Comment (barchiso) [reply Lesson_07/homework_07.py]: Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d

```
if number <= 1:
        return False
    if any(number % i == 0 for i in range(2, int(number ** 0.5) + 1)):
        return False
    return True
```

File: Lesson_07/homework_07.py (hunk 3 above)
Comment (barchiso) [reply Lesson_07/homework_07.py]: Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d

` return sep.join(str(arg) for arg in args)
`

File: Lesson_07/homework_07.py (hunk 2 above)
Comment (barchiso) [reply Lesson_07/homework_07.py]: Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d

`    return sum(number for number in numbers_list if number % 2 == 0)
`

File: Lesson_07/homework_07.py (hunk 1 above)
Comment (barchiso) [reply Lesson_07/homework_07.py]: Updated in commit: f61bf23d85eca83d002c0c3ff63679065222887d

`    return str1.find(str2) if str2 in str1 else -1
`

File: Lesson_07/homework_07.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    """First entry of the second line into the first line.
+
+    Args:
+        str1 (string): First string
+        str2 (string): Second string
+
+    Returns:
+        int: Index of the first entry
+    """
+    return str1.find(str2) if str2 in str1 else -1
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]:
```suggestion
    return str1.find(str2)
```
find returns -1 if nothing found -> see help(str.find)

File: Lesson_07/homework_07.py [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        number (integer): any number
+
+    Returns:
+        boolean: True or False
+    """
+    if number <= 1:
+        return False
+    if any(number % i == 0 for i in range(2, int(number ** 0.5) + 1)):
+        return False
+    return True
```
Comment (antohaUa) [inline Lesson_07/homework_07.py]:
```suggestion
   return (number > 1 and not any(number % i == 0 for i in range(2, int(number ** 0.5) + 1)))
```

**Review by antohaUa — APPROVED** [review]

**Comment by barchiso** [comment]
Updated

**Pull request merged**